core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -15,191 +15,173 @@
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
+    return IMPL.gettimeofdayAvailable();
   }
 
   /**
    * The current time in microseconds, as returned by libc.gettimeofday(); can only be used if
    * {@link #isCurrentTimeMicrosAvailable()} is true.
    */
   public static long currentTimeMicros() {
-    if (!isCurrentTimeMicrosAvailable()) {
-      throw new IllegalStateException(
-          "Native call not available. "
-              + "Check isCurrentTimeMicrosAvailable() before calling this method.");
-    }
-    LibCLoader.Timeval tv = new LibCLoader.Timeval(LibCLoader.LIB_C_RUNTIME);
-    int res = LibCLoader.LIB_C.gettimeofday(tv, null);
-    if (res != 0) {
-      throw new IllegalStateException("Call to libc.gettimeofday() failed with result " + res);
-    }
-    return tv.tv_sec.get() * 1000000 + tv.tv_usec.get();
+    return IMPL.gettimeofday();
   }
 
   public static boolean isGetProcessIdAvailable() {
-    try {
-      return PosixLoader.GET_PID_AVAILABLE;
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
+    return IMPL.getpidAvailable();
   }
 
   public static int getProcessId() {
-    if (!isGetProcessIdAvailable()) {
-      throw new IllegalStateException(
-          "Native call not available. "
-              + "Check isGetProcessIdAvailable() before calling this method.");
-    }
-    return PosixLoader.POSIX.getpid();
+    return IMPL.getpid();
   }
 
   /**
-   * Returns {@code true} if JNR {@link Platform} class is loaded, and {@code false} otherwise.
-   *
-   * @return {@code true} if JNR {@link Platform} class is loaded.
-   */
-  public static boolean isPlatformAvailable() {
-    try {
-      return PlatformLoader.PLATFORM != null;
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
-  }
-
-  /**
-   * Returns the current processor architecture the JVM is running on, as reported by {@link
-   * Platform#getCPU()}.
+   * Returns the current processor architecture the JVM is running on. This value should match up to
+   * what's returned by jnr-ffi's Platform.getCPU() method.
    *
    * @return the current processor architecture.
-   * @throws IllegalStateException if JNR Platform library is not loaded.
    */
   public static String getCPU() {
-    if (!isPlatformAvailable())
-      throw new IllegalStateException(
-          "JNR Platform class not loaded. "
-              + "Check isPlatformAvailable() before calling this method.");
-    return PlatformLoader.PLATFORM.getCPU().toString();
+    return CPU.toString();
   }
 
-  /**
-   * If jnr-ffi is not in the classpath at runtime, we'll fail to initialize the static fields
-   * below, but we still want {@link Native} to initialize successfully, so use an inner class.
+  /* The remainder of this class is copied from jnr.ffi.Platform in jnr-ffi version 2.1.10.
+   * We copy it manually here in order to avoid introducing an extra dependency merely for the sake of
+   * evaluating some system properties.
```

Comment:
In isolation this comment looks a bit odd.  We already have JnrNativeImpl which hides the dependency on jnr-posix... why not do something similar with jnr-ffi and avoid copying code?  This suggestion breaks down when we consider adding Graal support in JAVA-2663.  In that case we'll still want to provide this functionality for _both_ impls but ideally we wouldn't mix jnr-ffi and the Graal impl.  With this in mind the cleanest answer was to promote this functionality to the Native level and implement it by copyig code.

Reply:
I suppose we... could just return UNKNOWN, but I'm not sure why we'd take that path.  The functionality to determine the return val of getCpu() isn't based on JNR; it's a general manipulation of standard system properties.  Why _wouldn't_ we support that (and support it in the same way) for all NativeImpls?

Reply:
Ok, fair enough.